modules/library/src/main/scala/zio/elasticsearch/ElasticRequest.scala

```diff
@@ -611,7 +612,7 @@ object ElasticRequest {
   private[elasticsearch] final case class Search(
     excluded: Chunk[String],
     included: Chunk[String],
-    index: IndexName,
+    index: String,
```

Comment:
Instead of `String` put `IndexSelector` (GetById, Refresh, Search), and then in HttpExecutor make `String` from it.